### ISSUE TITLE
Close buttons' icons in clr-alert and clr-modal should have title set

### ISSUE THREAD

**Issue by immihaylova** [vmware/clarity#4508]
This is an issue with accessibility. We got a report from AccessLevel where they note that icons inside a button or a link that have no adjacent text should have tooltip and aria-label set.

## How to reproduce

Use clr-alert/clr-modal and check their close button icon accessibility properties (screenshots attached).

## Expected behavior

The clr-icon element inside the close button should have its title property set. This will add a toolip and an aria-label for the icon.

## Versions

* Angular: 9
* Node: 10
* Clarity: 3.0
**Device:** 

* Type: Desktop
* OS: Win10
* Browser: Chrome
* Version 81

Alert:
![clr_alert_close_btn](https://user-images.githubusercontent.com/43816378/80216354-97c00d00-8646-11ea-8efa-06887dc319a3.png)

Modal:
![clr_modal_close_btn](https://user-images.githubusercontent.com/43816378/80216393-a8708300-8646-11ea-8465-7710ae863335.png)

**Comment by coryrylan** [comment]
Aria label and title attr should not be on an icon directly when inside an icon only button. The fix for this is to put the title on the surrounding button to match with the `aria-label` as that is the action element we are describing. We can update the example here https://clarity.design/icons/accessibility

```html
<button class="btn btn-icon" aria-label="main menu" title="main menu">
  <clr-icon shape="bars"></clr-icon>
</button>
```

**Comment by immihaylova** [comment]
> Aria label and title attr should not be on an icon directly when inside an icon only button. The fix for this is to put the title on the surrounding button to match with the `aria-label` as that is the action element we are describing. We can update the example here https://clarity.design/icons/accessibility
> 
> ```
> <button class="btn btn-icon" aria-label="main menu" title="main menu">
>   <clr-icon shape="bars"></clr-icon>
> </button>
> ```
> 
> Related issue #4454

Thanks, Cory.

It seems to me that if you go this way, you will also need to add 'aria-hidden=true' to the clr-icon. Otherwise the svg inside it will be visible to screen readers and may cause confusion.

<button class="btn btn-icon" aria-label="main menu" title="main menu">
   <clr-icon shape="bars" aria-hidden=true></clr-icon>
</button>

**Comment by mathisscott** [comment]
@immihaylova 
Thanks for the suggestion. Clarity's a11y support is driven by the a11y team at VMware. Because there is so much left up to interpretation as regards implementing features that meet the a11y guidelines these days, we rely on their expertise when we hit differing opinions between vendors or developers on compliance. What is in Clarity right now has already been vetted by the Clarity a11y team. 

While I believe generally that there is always room for improvement, I also believe it would be unwise for us to take on enhancements without their guidance. Especially if new proposals run counter to the previous guidance we have received from them.

Pinging @chlane and @sheribyrne to help us here.

**Comment by chlane** [comment]
The need for an accessible name for the close buttons in alerts was addressed in https://github.com/vmware/clarity/issues/3498. But the usage of the title attribute is not recommended. Although the title attribute is included in the accessible name computation it is not accessible. Sighted keyboard-only users will not be able to activate the visual tooltip that is displayed when the element with the  title attribute is moused over because they cannot use a mouse and these tooltips do not appear on focus.

My recommendation when originally reporting this issue was to provide a mechanism for the developer using clarity to pass in a specific accessible name for the close button for any alert. Simply using the text 'close' is not specific enough since there may be similar buttons on the page for other controls.

**Comment by chlane** [comment]
Also for any icon using focusable="false" we should remove role="img"  and add aria-hidden="true" to clean up the accessibility tree.

**Comment by gnomeontherun** [comment]
Clarity cares deeply about accessibility, and we work with our accessibility team to identify and triage issues. We have an internal tracking system to identify and verify accessibility issues, which has captured this already. Some issues with Clarity Angular are difficult to solve without causing side effects or breaking changes, and we have to balance concerns with fixing issues and breaking changes. As we build out Clarity Core, we are able to address these root cause accessibility issues and recommend adopting the Clarity Core implementations as they become available. We are closing this issue as we have the issues tracked internally. Please follow our development and releases to see when we release relevant Clarity Core components to make this possible.
